debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/testhelper/RestExtensionTestInfrastructure.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.testing.testcontainers.testhelper;
+
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import org.awaitility.Awaitility;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.MSSQLServerContainer;
+import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.startupcheck.MinimumDurationRunningStartupCheckStrategy;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.lifecycle.Startable;
+import org.testcontainers.utility.DockerImageName;
+
+import io.debezium.testing.testcontainers.Connector;
+import io.debezium.testing.testcontainers.ConnectorConfiguration;
+import io.debezium.testing.testcontainers.ConnectorResolver;
+import io.debezium.testing.testcontainers.DebeziumContainer;
+import io.debezium.testing.testcontainers.MongoDbReplicaSet;
+import io.debezium.testing.testcontainers.util.MoreStartables;
+
+public class RestExtensionTestInfrastructure {
```

Comment:
Why should this interfere? The only touch point can be TCP ports (which are/should be dynamically managed by Docker) or components which use broadcasts to discover the cluster (should not be possible with Docker network isolation until you not expose everything to the local physical network).
Like I said above, I think at the moment we have to accept that we have a mix of old an new things for historical reasons and no time to spent the effort to migrate everything now. But I am also against keeping the legacy way of testing because then we make the mountain higher and higher that we have to walk eventually.

Reply:
So what added value you get with TC databases compared to what we have? Which cannot be gradually upgraded with the rest (say connector by connector -- the same way as it was done with MongoDB, for which there were additional reasons to switch in order to support sharded clusters and proper replica sets).

Reply:
Also what about the complexity and the fact that I don't think this class shoudl exists within the testcontainers module.

Reply:
Should the class be named "TestStackForRestExtensions"  or something similar? It is a template or a composition for some testcontainers meant for testing the generic and connector-specific REST extensions.

Reply:
This class should not really belong in `debezium-testing`

Reply:
@jpechane  Can you suggest a better place? What's wrong with my previous point of view about that this describes a tech stack for containers in `debezium-testing-testcontainers`? `debezium-testing-testcontainers` can produce the containers but also full test-stacks, no?

Reply:
I filed https://issues.redhat.com/browse/DBZ-6947 for tackling those concerns. @jpechane @jcechace  Could you please add some description to the issue about what exactly you would like to see/achieve?